### PR TITLE
Fix/one variable model

### PR DESCRIPTION
**Pull request by lzampier**
## Description

The current version of anemoi fails when models with just one prognostic variable are trained because of some .squeeze statements that reduce arrays to scalars. I use .reshape(-1) instead. This seems to fix the issue in my tests and still works with more complex models with multiple variables.

## What problem does this change solve?

Enables models with one prognostic variable to be trained in anemoi-train. This is arguably an edge case, which should be enabled nevertheless.

## What issue or task does this change relate to?

This change addresses issue #477. 

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md)

### CODE REVIEW

File: training/src/anemoi/training/diagnostics/plots.py [hunk 1]
```diff
@@ -229,15 +229,15 @@ def plot_power_spectrum(
     grid_pc_lon, grid_pc_lat = np.meshgrid(regular_pc_lon, regular_pc_lat)
 
     for plot_idx, (variable_idx, (variable_name, output_only)) in enumerate(parameters.items()):
-        yt = y_true[..., variable_idx].squeeze()
-        yp = y_pred[..., variable_idx].squeeze()
+        yt = (y_true if y_true.ndim == 1 else y_true[..., variable_idx]).reshape(-1)
```
Comment (anaprietonem) [inline training/src/anemoi/training/diagnostics/plots.py:232]: @lzampier - why do you need the if here - could just do y_true[..., variable_idx]).reshape(-1)? and same for x

Comment (lzampier) [reply training/src/anemoi/training/diagnostics/plots.py:232]: Probably yes. I was overcautious. Let me test later.

Comment (anaprietonem) [reply training/src/anemoi/training/diagnostics/plots.py:232]: Great, apart from that I think it could be good if we could have a test added here for the `print_variable_scaling` where we can see if it works fine for one and more than one variable https://github.com/ecmwf/anemoi-core/blob/5a311574b9fb3ab6d610e8e80fc3d91eab061a82/training/src/anemoi/training/losses/utils.py

could you give that a go? I can review later or if you have doubts just drop me a message!

Comment (lzampier) [reply training/src/anemoi/training/diagnostics/plots.py:232]: I checked, and yes we need the `(y_true if y_true.ndim == 1 else` part, otherwise the plotting fails. Regarding the tests, I have done an attempt that you can see in the newest commit. Let me know if you think it is appropriate. The behavior is a bit strange to test, so maybe you have a better idea.

File: training/tests/unit/train/test_print_variable_scaling.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# (C) Copyright 2024 Anemoi contributors.
```
Comment (anaprietonem) [inline training/tests/unit/train/test_print_variable_scaling.py]: header should be 2025

File: training/tests/unit/train/test_print_variable_scaling.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    names = ["swh", "mwp"]
+    indices = _fake_indices(names)
+    shapes = [(2,), (1, 2), (2, 1), (1, 1, 2)]  # all contain exactly 2 elements
+
+    for shape in shapes:
+        base = np.array([1.0, 2.5]).reshape(shape)
+        loss = _FakeLoss(base)
+
+        handler = _ListHandler()
+        old_level = LOGGER.level
```
Comment (anaprietonem) [inline training/tests/unit/train/test_print_variable_scaling.py]: why do we need this for the LOGGER?

File: training/tests/unit/train/test_print_variable_scaling.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    payload = message.removeprefix(prefix)
+    # The function ends with a trailing ", " — filter empty tail
+    parts = [p for p in payload.split(", ") if p]
+    return len(parts)
+
+
+# ------------------------------ Tests ---------------------------------------
+
+
+def test_print_variable_scaling_single_var_flattens_but_not_scalar() -> None:
```
Comment (anaprietonem) [inline training/tests/unit/train/test_print_variable_scaling.py]: my proposal for the test would be to reuse the tests/unit/train/test_loss_scaling.py -test_variable_loss_scaling_vals 

There we can get the data_indices and loss and then call `print_variable_scaling(loss, indices)` - checking that the  test passes with one and many variables

**Review by anaprietonem — APPROVED** [review]
Nice fix, and tests have been included! GTM
